### ISSUE TITLE
High CPU usage when using multiple open editors (js/ts)

### ISSUE THREAD

**Issue by Madd0g** [Microsoft/vscode#47817]
- VSCode Version: 1.22.1
- OS Version: OSX 10.11.6

Steps to Reproduce:

1. Have multiple javascript/typescript projects open in multiple windows (I might have up to 6-7 open at a time), but this happens with 2 as well.
2. Get very high CPU usage from "Code Helper", switch between all the open VSCode windows and it stops, until the next time it happens.

Notes:

* Landing on "the right" VSCode window always stops the problem. I think it's pretty random, any window could cause the problem. But bringing focus to it stops the problem.

* I have typescript watchers, webpack watchers, nodemon, stuff like that, running in the background, watching the same files the various vscode windows have open as well. I also have `"files.autoSave": "onFocusChange",` in the settings.

* After every version upgrade, I get the update notification a lot of times, in different instances of vscode, unexpectedly. For example I open a project and get the notification and dismiss it, a few days later I get it again and notice in the mac dock, there are two vscode icons instead of one (so 2 version running in parallel). Don't know if it's related, but here's a screenshot of how it looks right now (2 instances, one without icon):

<img width="797" alt="image" src="https://user-images.githubusercontent.com/1171003/38727785-d6fb8044-3ec2-11e8-96d1-1fa922bca343.png">

I found that one of the processes is launched from `~/Library/Caches/com.microsoft.VSCode.ShipIt/update.reIUiIT/Visual Studio Code.app/Contents/MacOS/Electron`.

**Comment by Madd0g** [comment]
```
Version:          Code 1.22.2 (3aeede733d9a3098f7b4bdc1f66b63xxxxxxx, 2018-04-12T16:32:53.389Z)
OS Version:       Darwin x64 15.6.0
CPUs:             Intel(R) Core(TM) i7-4850HQ CPU @ 2.30GHz (8 x 2300)
Memory (System):  16.00GB (0.94GB free)
Load (avg):       3, 3, 3
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron

CPU %	Mem MB	   PID	Process
   75	   115	 64590	code main
    0	   115	 64591	   gpu-process
    0	    66	 64596	   shared-process
   75	   279	 69465	   window (MY-FILE.js (Working Tree) — MY_PROJECT1)
    0	   131	 69468	     extensionHost
    0	   147	 69487	       electron_node electronForkStart.js tsserver.js
    0	    33	 69515	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    33	 69488	       electron_node eslintServer.js
    0	    33	 69469	     watcherService
    0	    33	 69492	     searchService
    0	   197	 78737	   window (MY-FILE2.ts — MY_TS_PROJECT2)
    0	   115	 78738	     extensionHost
    0	    66	 78761	       electron_node tslintServer.js
    0	    98	 78762	       electron_node electronForkStart.js tsserver.js
    0	    33	 78786	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    33	 78810	       electron_node jsonServerMain.js
    0	    33	 78739	     watcherService
    0	    33	 78766	     searchService

Workspace Stats:
|  Window (MY-FILE.js (Working Tree) — MY_PROJECT1)
|    Folder (MY_PROJECT1): 158 files
|      File types: js(68) html(61) xml(9) json(6) css(4) png(2) md(1)
|                  babelrc(1) iml(1) DS_Store(1)
|      Conf files: package.json(1)
|  Window (MY-FILE2.ts — MY_TS_PROJECT2)
|    Folder (MY_TS_PROJECT2): 213 files
|      File types: html(67) ts(56) js(28) map(21) json(12) xml(6) yml(3) md(2)
|                  DS_Store(2) css(2)
|      Conf files: tsconfig.json(2) launch.json(1) tasks.json(1)
|                  package.json(1) tslint.json(1)
```

**Comment by vesper8** [comment]
Just updated to 1.22.2 just now and my Code Helper has been at 100% CPU usage for the last 15 minutes (OSX High Sierra)

I have two editors opened

And here's the results of code --status

```
[64060:0416/193742.413138:WARNING:dns_config_service_posix.cc(154)] dns_config has unhandled options!

Version:          Code 1.22.2 (3aeede733d9a3098f7b4bdc1f66b63b0f48c1ef9, 2018-04-12T16:32:53.389Z)
OS Version:       Darwin x64 17.5.0
CPUs:             Intel(R) Core(TM) i7-4850HQ CPU @ 2.30GHz (8 x 2300)
Memory (System):  16.00GB (0.04GB free)
Load (avg):       3, 4, 4
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/_DEV/Visual Studio Code.app/Contents/MacOS/Electron
CPU %   Mem MB     PID  Process
    0       82   62221  code main
    0       33   62222     gpu-process
    0       66   62224     shared-process
    0      229   62545     window (User.php — omitted)
    0      197   62547       extensionHost
    0       33   62559         electron_node jsonServerMain.js
    0     1098   62577         electron_node server.js
    0       16   62578         php /Users/vesperknight/.vscode/extensions/felixfbecker.php-intellisense-2.3.1/vendor/felixfbecker/language-server/bin/php-language-server.php --tcp=127.0.0.1:49280 --memory-limit=4095M
    0     1049   62579           /usr/local/Cellar/php72/7.2.2_13/bin/php -c /private/var/folders/fq/k04z_8kj3pl1q15dhrzpfdzr0000gn/T/gjTzWz /Users/vesperknight/.vscode/extensions/felixfbecker.php-intellisense-2.3.1/vendor/felixfbecker/language-server/bin/php-language-server.php --tcp=127.0.0.1:49280 --memory-limit=4095M
    0       49   62548       watcherService
    0       66   62561       searchService
    0      164   62580     window (Teams.vue — pocvue)
    0       33   62583       extensionHost
    0      147   62588         electron_node vueServerMain.js
    0       33   62590         electron_node eslintServer.js
    0       33   62591         electron_node server.js
    0       16   62584       watcherService
    0       33   62595       searchService
Workspace Stats:
|  Window (User.php — omitted)
|    Folder (omitted): more than 20000 files
|      File types: php(11549) js(5184) png(637) css(331) stub(271) less(232)
|                  md(194) json(186) scss(181) DS_Store(141)
|      Conf files: package.json(8) makefile(7) gulp.js(3) grunt.js(2)
|  Window (Teams.vue — pocvue)
|    Folder (pocvue): 300 files
|      File types: js(77) vue(53) png(45) map(42) css(18) DS_Store(14)
|                  json(10) jpg(6) svg(6) ico(4)
|      Conf files: settings.json(1) jsconfig.json(1) package.json(1)
```

**Comment by Madd0g** [comment]
I only worked with ts/js in vscode. I'll try the insiders build.

If I had to put my finger on when it happens, it usually happens when I'm "abandoning" an editor, make a little change in one window and then use another vscode window for a while. Like if feels insulted that I left it.

It happens every day but only a few times a day, I don't know if I could survive without JS/TS language features for long enough to be able to tell if it solved the problem completely.

For full disclosure sake, I had exactly the same trouble with other editors for years (atom & webstorm, even on other macs and other codebases), so I would definitely put my finger on tsserver because if I'm not mistaken they all used it at some point.

**Comment by fedorg** [comment]
**Code 1.23.0** 
My system is now affected as well. It typically happens when I have multiple VSCode windows open, however it is not uncommon with just one window. ESLint extension is disabled in both workspaces.


<details><summary>settings.json (both workspaces)</summary>

```
{
    "files.watcherExclude": { "**/**": true }
}
```
</details>

<details><summary>code --status</summary>

```
ATTENTION: option value of option force_s3tc_enable ignored.
Failed to watch /home/user/.config/Code/User for changes (Error: watch /home/user/.config/Code/User ENOSPC)

Version:          Code 1.23.0 (7c7da59c2333a1306c41e6e7b68d7f0caa7b3d45, 2018-05-03T15:23:19.356Z)
OS Version:       Linux x64 4.13.0-39-generic
CPUs:             Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2693)
Memory (System):  15.55GB (0.45GB free)
Load (avg):       1, 1, 1
VM:               0%
Screen Reader:    no
Process Argv:     /usr/share/code/code --unity-launch
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    disabled_software
                  rasterization:                disabled_software
                  video_decode:                 unavailable_software
                  video_encode:                 enabled
                  vpx_decode:                   unavailable_software
                  webgl:                        enabled
                  webgl2:                       enabled
CPU %	Mem MB	   PID	Process
    2	   111	 30220	code main
    0	    16	 30224	   zygote
    4	   207	 30263	     window (settings.json - flava - Visual Studio Code)
    1	    64	 30348	       extensionHost
    2	   127	 30379	         electron_node electronForkStart.js tsserver.js 
    1	    48	 30417	           electron_node typingsInstaller.js tsserver.js typesMap.js 
    0	    64	 30773	         electron_node jsonServerMain.js 
    0	    48	 30349	       watcherService
    0	    48	 30383	       searchService
    1	    96	 30267	     shared-process
    2	   191	 31219	     window (index.js - antd-admin - Visual Studio Code)
    1	    64	 31252	       extensionHost
   41	   207	 31283	         electron_node electronForkStart.js tsserver.js 
    0	    48	 31253	       watcherService
    0	    48	 31278	       searchService
    1	    80	 30249	   gpu-process
Workspace Stats:
|  Window (settings.json - flava - Visual Studio Code)
|    Folder (flava): 23 files
|      File types: jsx(8) json(4) html(2) less(2) css(1) eslintignore(1)
|                  gitignore(1) js(1) editorconfig(1) webpackrc(1)
|      Conf files: launch.json(1) settings.json(1) package.json(1)
|  Window (index.js - antd-admin - Visual Studio Code)
|    Folder (antd-admin): 215 files
|      File types: js(121) less(38) svg(20) json(12) jpg(4) md(3) gif(2)
|                  eot(2) ttf(2) woff(2)
|      Conf files: package.json(10) settings.json(1) webpack.config.js(1)
</details>

**Comment by Madd0g** [comment]
@fedorg - my issue specifically mentions multiple windows because that's how I can easily classify this as a bug - simply activating one of the inactive windows stop the CPU from going crazy, so it couldn't have been doing anything important :)

I have been using the insiders build for over a week now (I started with just the must-have extensions and over time installed all the ones I had in the previous version) and this problem only happened once since I switched! I'm pretty happy not being worried about it! I will be on the lookout, but for now I feel like the problem is gone.

Also noticed some great improvements with the insiders build in general (vue files finally have proper autocompletion, it never worked for me before).

Thanks

**Comment by fedorg** [comment]
I've installed the latest vscode-insiders 1.24.0 and the CPU usage is even worse. It's not quite the same because the process that is hogging the CPU is now a child of what the hogging process would be on 1.23.0. It looks very similar to https://github.com/Microsoft/TypeScript/issues/22702 with both parent and child processes hogging the CPU:

![image](https://user-images.githubusercontent.com/4352852/39624607-9a49d9e8-4fa2-11e8-982a-86aa58aca838.png)

<details><summary>code-insiders --status</summary>

```
ATTENTION: option value of option force_s3tc_enable ignored.

Version:          Code - Insiders 1.24.0-insider (2404210629c744e6237a14d7b5fa852e24c6e898, 2018-05-04T08:06:25.006Z)
OS Version:       Linux x64 4.13.0-39-generic
CPUs:             Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2563)
Memory (System):  15.55GB (0.30GB free)
Load (avg):       2, 1, 2
VM:               0%
Screen Reader:    no
Process Argv:     /usr/share/code-insiders/code-insiders --unity-launch
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    disabled_software
                  rasterization:                disabled_software
                  video_decode:                 unavailable_software
                  video_encode:                 enabled
                  vpx_decode:                   unavailable_software
                  webgl:                        enabled
                  webgl2:                       enabled
CPU %	Mem MB	   PID	Process
    4	   111	  3670	code-insiders main
    0	    16	  3673	   zygote
    0	    80	  3721	     shared-process
    5	   223	  3929	     window (users.js - antd-admin - Visual Studio Code - Insiders)
    0	    64	  3958	       extensionHost
   21	   191	  4057	         electron_node electronForkStart.js tsserver.js 
   43	    80	  4065	           electron_node typingsInstaller.js tsserver.js typesMap.js 
    0	    48	  3961	       watcherService
    0	    48	  4019	       searchService
    3	    80	  4922	     shared-process
    2	    80	  3699	   gpu-process
Workspace Stats:
|  Window (users.js - antd-admin - Visual Studio Code - Insiders)
|    Folder (antd-admin): 215 files
|      File types: js(121) less(38) svg(20) json(12) jpg(4) md(3) gif(2)
|                  eot(2) ttf(2) woff(2)
|      Conf files: package.json(10) webpack.config.js(1) settings.json(1)

```

</details>

**Comment by Madd0g** [comment]
After many things broke in 1.24.0 I had to go back to regular VSCode for a while and even though it's now at the same version as the insiders version I used (1.23.x), the CPU problem is happening very often again :(

**Comment by ndrake** [comment]
I've also been experiencing similar behavior with multiple code windows open.

code --status
 
```
Version:          Code 1.23.1 (d0182c3417d225529c6d5ad24b7572815d0de9ac, 2018-05-10T16:03:31.083Z)
OS Version:       Darwin x64 17.5.0
CPUs:             Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2900)
Memory (System):  16.00GB (0.21GB free)
Load (avg):       3, 3, 3
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron .
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    enabled
                  rasterization:                enabled
                  video_decode:                 enabled
                  video_encode:                 enabled
                  vpx_decode:                   enabled
                  webgl:                        enabled
                  webgl2:                       enabled

CPU %	Mem MB	   PID	Process
    0	   131	 49537	code main
    0	    98	 49564	   window (app/modules/notebook/notebook-controller.js — PROJECT1)
    0	    33	 49568	     extensionHost
    0	    16	 49906	       electron_node eslintServer.js
    0	    33	 49922	       electron_node electronForkStart.js tsserver.js
    0	    16	 50065	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    16	 49569	     watcherService
    0	    16	 49571	     terminal
    0	     0	 49573	       /bin/zsh -l
    0	    16	 50208	     searchService
    0	    49	 49565	   shared-process
    0	   164	 53871	   window (package.json — PROJECT2)
    0	    49	 53872	     extensionHost
   98	    98	 53875	       electron_node electronForkStart.js tsserver.js
    0	    16	 53889	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    33	 53876	       electron_node eslintServer.js
    0	    33	 54275	       electron_node jsonServerMain.js
    0	   115	 53873	     watcherService
    0	    33	 53895	     searchService
    0	   131	 53942	   window (package.json — PROJECT3)
    0	    33	 53943	     extensionHost
    0	    33	 53949	       electron_node jsonServerMain.js
    0	    33	 53963	       electron_node eslintServer.js
    0	    33	 53964	       electron_node electronForkStart.js tsserver.js
    0	    16	 53969	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    16	 53944	     watcherService
    0	    16	 53946	     searchService

Workspace Stats:
|  Window (app/modules/notebook/notebook-controller.js — PROJECT1)
|    Folder (PROJECT1): 5605 files
|      File types: png(2744) svg(1244) js(637) json(281) scss(270) html(195)
|                  ttf(47) woff(47) eot(47) ico(14)
|      Conf files: settings.json(7) gulp.js(2) package.json(1)
|  Window (package.json — PROJECT2)
|    Folder (PROJECT2): 9732 files
|      File types: js(1591) flat(1524) class(1157) dex(1070) png(825)
|                  json(656) md(335) java(278) scss(259) h(247)
|      Conf files: package.json(52) settings.json(6) grunt.js(5) gulp.js(4)
|                  webpack.config.js(1) tsconfig.json(1) makefile(1) csproj(1)
|                  sln(1)
|  Window (package.json — PROJECT3)
|    Folder (PROJECT3): 5161 files
|      File types: js(1108) png(996) json(383) h(345) svg(316) scss(286)
|                  md(273) java(170) html(162) woff(124)
|      Conf files: package.json(46) settings.json(7) gulp.js(3) grunt.js(1)
|                  tsconfig.json(1) csproj(1) sln(1)
```

**Comment by brookback** [comment]
Hi!

I've been experiencing this as well on 1.23.1.

- I've got a single window open, with four tabs. All JS (a Meteor project).
- I have to reboot Code to make the CPU go down.
- Running ESLint
- It's usually a single `Code Helper` process going crazy, steadily on >100% CPU:
   ![skarmavbild 2018-05-21 kl 10 40 33](https://user-images.githubusercontent.com/30257156/40298503-abc9ba00-5ce3-11e8-8e10-6fff64f308ea.png)

```
Version:          Code 1.23.1 (d0182c3417d225529c6d5ad24b7572815d0de9ac, 2018-05-10T16:03:31.083Z)
OS Version:       Darwin x64 17.3.0
CPUs:             Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz (4 x 3100)
Memory (System):  16.00GB (0.04GB free)
Load (avg):       3, 3, 3
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron .
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    enabled
                  rasterization:                enabled
                  video_decode:                 enabled
                  video_encode:                 enabled
                  vpx_decode:                   enabled
                  webgl:                        enabled
                  webgl2:                       enabled

CPU %	Mem MB	   PID	Process

Workspace Stats:
|  Window (domain-signup.js — lookback-site)
|    Folder (lookback-site): 14625 files
|      File types: json(5240) cache(3401) js(1997) png(1970) html(310)
|                  scss(199) coffee(198) svg(193) wt(161) map(145)
|      Conf files: package.json(14) makefile(3) grunt.js(3) settings.json(2)
|                  launch.json(1)
|      Launch Configs: node(2)
```

**Comment by GotJimmy** [comment]
Same issue here in 1.24 using TypeScript and ReactXP. Very easy to duplicate: try to invoke IntelliSense code completion in strokeWidth={this. } and one Code Helper goes to 100% each time:

`<svg>
<line className='metka' x1='180' y1='14' x2='180' y2='26' transform='rotate(226 180 180)' stroke={this.props.tickColor} strokeWidth={this.} />
</svg>`

Here's the output from code --status:

> Version:          Code 1.24.0 (6a6e02cef0f2122ee1469765b704faf5d0e0d859, 2018-06-06T17:30:30.113Z)
OS Version:       Darwin x64 16.7.0
CPUs:             Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz (8 x 2300)
Memory (System):  8.00GB (0.15GB free)
Load (avg):       3, 3, 3
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    enabled
                  rasterization:                enabled
                  video_decode:                 enabled
                  video_encode:                 enabled
                  vpx_decode:                   enabled
                  webgl:                        enabled
                  webgl2:                       enabled

> CPU %	Mem MB	   PID	Process
    1	   115	 21311	code main
    1	   147	 21312	   gpu-process
    1	   254	 21313	   window (client (Workspace))
   99	   180	 21329	     extensionHost
    0	   188	 21381	       electron_node electronForkStart.js tsserver.js 
    0	    49	 21395	         electron_node typingsInstaller.js tsserver.js typesMap.js 
    0	    41	 21383	       electron_node tslintServer.js 
    0	    49	 21429	       electron_node jsonServerMain.js 
    0	    33	 21330	     watcherService
    0	    33	 21332	     terminal
    0	     0	 21333	       /usr/local/bin/fish -l
    0	    49	 21394	     searchService
    0	   319	 21314	   window (server (Workspace))
    0	   115	 21327	     extensionHost
    0	    90	 21340	       electron_node electronForkStart.js tsserver.js 
    0	    49	 21376	         electron_node typingsInstaller.js tsserver.js typesMap.js 
    0	    41	 21341	       electron_node tslintServer.js 
    0	    49	 21405	       electron_node jsonServerMain.js 
    0	    33	 21328	     watcherService
    0	    41	 21350	     searchService
    0	   106	 21315	   shared-process

> Workspace Stats: 
|  Window (client (Workspace))
|  Window (server (Workspace))
|    Folder (common): 5 files
|      File types: ts(4) DS_Store(1)
|      Conf files:
|    Folder (common): 5 files
|      File types: ts(4) DS_Store(1)
|      Conf files:
|    Folder (server): 70 files
|      File types: log(43) DS_Store(6) js(5) json(5) ts(4) gz(2) html(1)
|                  gitignore(1) lock(1) map(1)
|      Conf files: package.json(1) tsconfig.json(1) webpack.config.js(1)
|                  launch.json(1)
Unable to parse /Users/cpg/dev/typescript/_/vulcansoft.com/server/.vscode/launch.json
|    Folder (client): 4204 files
|      File types: json(3369) h(211) js(168) pak(165) map(80) tsx(42)
|                  DS_Store(19) plist(19) ts(18) svg(14)
|      Conf files: package.json(2) tsconfig.json(1) launch.json(1)
|                  settings.json(1) webpack.config.js(1)
Unable to parse /Users/cpg/dev/typescript/_/vulcansoft.com/client/.vscode/launch.json

**Comment by GotJimmy** [comment]
I found the culprit for the issue in my previous post: the extension "Auto Rename Tag" caused it. Uninstalled it and now everything's fine :)

**Comment by ndrake** [comment]
I'm still seeing this popup with 1.24.1.

```
Version:          Code 1.24.1 (24f62626b222e9a8313213fb64b10d741a326288, 2018-06-13T17:41:50.128Z)
OS Version:       Darwin x64 17.6.0
CPUs:             Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2900)
Memory (System):  16.00GB (0.30GB free)
Load (avg):       4, 4, 4
VM:               0%
Screen Reader:    no
Process Argv:     /Applications/Visual Studio Code.app/Contents/MacOS/Electron .
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    enabled
                  rasterization:                enabled
                  video_decode:                 enabled
                  video_encode:                 enabled
                  vpx_decode:                   enabled
                  webgl:                        enabled
                  webgl2:                       enabled

CPU %	Mem MB	   PID	Process
    0	    82	  4359	code main
    0	    82	  4384	   gpu-process
    0	   180	  4385	   window (app/components/select-text-to-learn/select-text-to-learn-popover.scss — WEB1)
    0	    16	  4389	     extensionHost
    0	     0	  4393	       electron_node eslintServer.js
    0	    16	  4395	       electron_node electronForkStart.js tsserver.js
    0	     0	  4416	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	     0	  4602	       electron_node htmlServerMain.js
    0	    16	  4623	       electron_node cssServerMain.js
    0	     0	  4390	     watcherService
    0	     0	  4599	     searchService
    0	    49	  4386	   shared-process
    0	    49	  5205	   window (app/components/popover/popover.scss — COMMON)
    0	     0	  5206	     extensionHost
    0	     0	  5209	       electron_node cssServerMain.js
    0	     0	  5234	       electron_node eslintServer.js
    0	    16	  5236	       electron_node electronForkStart.js tsserver.js
    0	     0	  5241	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	     0	  5249	       electron_node htmlServerMain.js
    0	     0	  5207	     watcherService
    0	     0	  5230	     searchService
    0	   246	 13044	   window (webpack.conf.js — WEB2)
    0	    82	 13045	     extensionHost
    0	    49	 13050	       electron_node eslintServer.js
   98	   197	 13055	       electron_node electronForkStart.js tsserver.js
    0	    33	 13071	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    33	 13076	       electron_node jsonServerMain.js
    0	    82	 13046	     watcherService
    0	    33	 13047	     searchService

Workspace Stats:
|  Window (app/components/select-text-to-learn/select-text-to-learn-popover.scss — WEB1)
|  Window (app/components/popover/popover.scss — COMMON)
|  Window (webpack.conf.js — WEB2)
|    Folder (WEB1): 4238 files
|      File types: png(2220) svg(1180) js(390) scss(156) html(143) json(30)
|                  ttf(24) eot(24) woff(24) gif(10)
|      Conf files: gulp.js(1) package.json(1) settings.json(1)
|    Folder (WEB2): 5680 files
|      File types: png(2789) svg(1244) js(648) json(285) scss(278) html(201)
|                  woff(47) ttf(47) eot(47) ico(14)
|      Conf files: settings.json(7) gulp.js(2) package.json(1)
|                  tsconfig.json(1)
|    Folder (COMMON): more than 20291 files
|      File types: png(7545) json(3796) js(3712) scss(1641) svg(877) html(816)
|                  woff(322) ttf(322) eot(322) DS_Store(219)
|      Conf files: settings.json(99) package.json(43)

```

**Comment by jsumners** [comment]
I am seeing this same issue after an update today. I can open a single project, e.g. `cd project-1 && code .`, and everything will be fine. As soon as I open a second project, `cd project-2 && code .`, the performance issue shows up. The second project window is *very* laggy, to the point of regularly prompting that it has stopped responding and wants to know if I want to "close, keep waiting, or reload". When I close the second project's window, things return to normal.

```
Version:          Code 1.25.0 (0f080e5267e829de46638128001aeb7ca2d6d50e, 2018-07-05T13:02:28.333Z)
OS Version:       Darwin x64 17.6.0
CPUs:             Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2900)
Memory (System):  16.00GB (0.04GB free)
Load (avg):       3, 2, 2
VM:               0%
Screen Reader:    no
Process Argv:     /Users/jsumners/Applications/Visual Studio Code.app/Contents/MacOS/Electron .
GPU Status:       2d_canvas:                    enabled
                  flash_3d:                     enabled
                  flash_stage3d:                enabled
                  flash_stage3d_baseline:       enabled
                  gpu_compositing:              enabled
                  multiple_raster_threads:      enabled_on
                  native_gpu_memory_buffers:    enabled
                  rasterization:                enabled
                  video_decode:                 enabled
                  video_encode:                 enabled
                  vpx_decode:                   enabled
                  webgl:                        enabled
                  webgl2:                       enabled

CPU %	Mem MB	   PID	Process
    2	    82	  1980	code main
    0	   164	  2001	   gpu-process
    0	   311	  2002	   window (offer.js — framework-data)
    0	    66	  2003	     extensionHost
    0	   344	  2149	       electron_node electronForkStart.js tsserver.js
    0	    16	  2150	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    16	  2153	       electron_node jsonServerMain.js
    0	    16	  2154	       electron_node htmlServerMain.js
    0	    16	  2156	       electron_node cssServerMain.js
    0	    16	  2157	       electron_node server.js
    0	    16	  2006	     watcherService
    0	    16	  2008	     searchService
    0	    16	  2027	     terminal
    0	     0	  2028	       /opt/local/bin/zsh -l
    0	    98	  2023	   shared-process
  100	  1753	  6065	   window (index.js — lambda-functions)
   15	   377	  6066	     extensionHost
   84	   786	  6072	       electron_node electronForkStart.js tsserver.js
   82	    82	  6090	         electron_node typingsInstaller.js tsserver.js typesMap.js
    0	    16	  6067	     watcherService
    0	    16	  6073	     searchService

Workspace Stats:
|  Window (index.js — lambda-functions)
|  Window (offer.js — framework-data)
|    Folder (framework-data): 221 files
|      File types: js(197) sql(16) md(3) json(2) eslintrc(1) gitignore(1)
|                  prettierrc(1)
|      Conf files: package.json(1) settings.json(1)
|    Folder (lambda-functions): 153 files
|      File types: js(69) json(27) html(12) env(11) md(7) pug(6) prettierrc(5)
|                  eslintrc(5) css(5) png(3)
|      Conf files: package.json(5) project.json(1) launch.json(1)
|                  settings.json(1)
|      Launch Configs: node(8)
```

**Comment by jsumners** [comment]
Downgrading to 1.24.1 "solves" this issue for me (https://code.visualstudio.com/updates/v1_24).

**Comment by mjbvz** [comment]
Closing this issue as it sounds like upgrading to a newer VS Code version resolved this for the original poster 

If you are still seeing perf issues, please open a new issue so that we can investigate
